src/panels/config/entities/entity-registry-settings.ts

```diff
@@ -459,6 +465,27 @@ export class EntityRegistrySettings extends SubscribeMixin(LitElement) {
               </ha-select>
             `
           : ""}
+        ${domain === "sensor" &&
+        // Allow customizing the precision for a sensor with numerical device class,
+        // a unit of measurement or state class
+        ((this._deviceClass &&
+          this._deviceClass !== "date" &&
+          this._deviceClass !== "enum" &&
+          this._deviceClass !== "timestamp") ||
```

Comment:
Could be written as `!["date", "enum", "timestamp"].includes(this._deviceClass)`